crates/router/src/core/payments/transformers.rs

```diff
@@ -2431,3 +2499,220 @@ impl ForeignFrom<CustomerDetails> for router_request_types::CustomerDetails {
         }
     }
 }
+
+#[cfg(feature = "v2")]
+impl ForeignFrom<api_models::payments::AmountDetails>
+    for hyperswitch_domain_models::payments::AmountDetails
+{
+    fn foreign_from(amount_details: api_models::payments::AmountDetails) -> Self {
+        Self {
+            order_amount: amount_details.order_amount.into(),
+            currency: amount_details.currency,
+            shipping_cost: amount_details.shipping_cost,
+            tax_details: Some(diesel_models::TaxDetails {
```

Comment:
How this always be Some?

Reply:
@sai-harsha-vardhan could you shed some light here?

Reply:
Yeah, this can't be always as Some. We can map this on `amount_details.order_tax_amount()`